### ISSUE TITLE
Empty Command Arguments

### ISSUE THREAD

**Issue by sparrell** [oasis-tcs/openc2-oc2ls#141]
LS-012 (https://www.oasis-open.org/apps/org/workgroup/openc2/email/archives/201811/msg00027.html):

> OpenC2-Command consists of 4 fields: {action, target, args, and actuator}.  The action and target fields are required, the args and actuator fields are optional.  Because each of the fields in the Args type is also optional, it is possible for a command with no command arguments to be serialized with or without an empty args field:
>
```
{ "action": "query",
  "target": {"features": []}}
```
> Or
```
{ "action": "query",
  "target": {"features": []},
  "args": {}}
```
> PROPOSED CHANGE:
>
> In Section 3.3.1.4, add a syntax requirement that Args is Type Map with [1..n] fields, and an equivalent Usage Note that Args must have at least one value.
> 
> JUSTIFICATION:
>
>Requiring args to be non-empty if present eliminates an unnecessary corner case and follows the “one way of doing things” design principle.

**Comment by jmbrule** [comment]
Agree with reviewer, need to explicitly state and we need to strive for the 'one way of doing things'

**Comment by romanojd** [comment]
At F2F, 01/2019, we agreed that, in general, if an optional list-property is present, it should contain at least one element.

**Comment by romanojd** [comment]
This is addressed by pull request #286.

We may want to apply this to the Response syntax. Wait until issue #236 is resolved.

**Comment by romanojd** [comment]
LSC Meeting, 3/19/2019: There were no objections to applying pull request #286 and closing this issue. I'm going to label this "future" because we may want to revisit applying this approach to the properties of Response.
